diff --git a/node_modules/@supercolony/typechain-compiler/index.ts b/node_modules/@supercolony/typechain-compiler/index.ts
index a025392..5e82f55 100644
--- a/node_modules/@supercolony/typechain-compiler/index.ts
+++ b/node_modules/@supercolony/typechain-compiler/index.ts
@@ -68,7 +68,7 @@ function main() {
 
 			logger.log(chalk.magenta(`======== Compiling ${contractName} ========`));
 
-			const cmd = `cargo +nightly contract ${isRelease ? "build --release" : "build"} --manifest-path ${tomlFile} ${config.skipLinting ? '--skip-linting' : ''}`;
+			const cmd = `cargo contract ${isRelease ? "build --release" : "build"} --manifest-path ${tomlFile} ${config.skipLinting ? '--skip-linting' : ''}`;
 
 			execSync(cmd);
 
@@ -77,9 +77,10 @@ function main() {
 
 
 		let targetInfo = tomlFiles.map((file: string) => {
+			let name = getContractNameFromToml(file);
 			return {
-				path: PathAPI.resolve(PathAPI.dirname(file), 'target', 'ink'),
-				name: getContractNameFromToml(file),
+				path: PathAPI.resolve('target', 'ink', name),
+				name,
 			};
 		});
 
